### PR TITLE
http-client: support CIDR in `no_proxy`

### PR DESCRIPTION
**Pull request by silverwind**
Support CIDR syntax in the `no_proxy` parsing of `http-client`.

The best "spec" on this topic is https://about.gitlab.com/blog/we-need-to-talk-no-proxy/ which shows the go and ruby parsers and probably others support CIDR in `no_proxy` and I have a specific use case where this broke when having `10.0.0.0/8` inside `no_proxy`.

Created with assistance of Claude Opus 5 and GPT 6 Astra.

### CODE REVIEW

File: packages/http-client/src/proxy.ts [hunk 1]
```diff
@@ -88,6 +91,19 @@ function isLoopbackAddress(host: string): boolean {
   )
 }
 
+function matchesCidr(ip: string, cidr: string): boolean {
+  const [network, prefix] = cidr.split('/')
+  if (!prefix || !isIP(ip)) return false
```
[the remark this answers is not in the document]
Comment (silverwind) [reply packages/http-client/src/proxy.ts]: fixed this edge case in https://github.com/actions/toolkit/pull/2495/commits/8c28864dbd77a1a2fa51638df01321cf340f396a.

File: packages/http-client/__tests__/proxy.test.ts [hunk 2]
```diff
@@ -183,6 +183,13 @@ describe('proxy', () => {
     expect(bypass).toBeTruthy()
   })
 
+  it('checkBypass matches IP hosts against no_proxy CIDR ranges and skips invalid ones', () => {
```
Comment (silverwind) [inline packages/http-client/__tests__/proxy.test.ts]:
```suggestion
  it('checkBypass supports CIDR in no_proxy', () => {
```
